code/game/machinery/doors/airlock.dm

```diff
@@ -1175,8 +1176,17 @@
 	delayed_close_requested = FALSE
 	if(safe)
 		CheckForMobs()
+	catchHair()
 	return TRUE
 
+/obj/machinery/door/airlock/proc/catchHair()
+	if(prob(10))
+		for(var/mob/living/carbon/human/H in orange(2, src))
```

Comment:
*music stops*

Reply:
I've gotten my hair stuck in doors that other people closed, and my hair is significantly shorter than floor length braids.

It makes more sense for it to be stuck on close than upon walk through.

Reply:
`orange()` means this will trigger for mobs through dense objects.